modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateSystemViewCommand.java

```diff
@@ -0,0 +1,161 @@
+import static org.apache.ignite.internal.catalog.CatalogParamsValidationUtils.validateIdentifier;
+import static org.apache.ignite.internal.util.CollectionUtils.nullOrEmpty;
+import static org.apache.ignite.lang.IgniteStringFormatter.format;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSystemViewDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableColumnDescriptor;
+import org.apache.ignite.internal.catalog.storage.NewSystemViewEntry;
+import org.apache.ignite.internal.catalog.storage.ObjectIdGenUpdateEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+import org.apache.ignite.lang.ErrorGroups.Common;
+
+/**
+ * Create system view command - creates or replaces a system view.
+ *
+ * <p>If a system with with the given name does not exists, this command adds this system view to {@code SYSTEM} schema.
+ * If a system with with the given name exists, this command replaces the existing system view.
+ */
+public class CreateSystemViewCommand extends AbstractCatalogCommand {
+
+    private final String name;
+
+    private final List<ColumnParams> columns;
+
+    /**
+     * Constructor.
+     *
+     * @param name View name.
+     * @param columns List of view columns.
+     */
+    CreateSystemViewCommand(String name, List<ColumnParams> columns) {
+        this.name = name;
+        this.columns = columns;
+
+        validate();
+    }
+
+    /**
+     * Returns a view name.
+     *
+     * @return View name.
+     */
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Returns a list of view columns.
+     *
+     * @return List of view columns.
+     */
+    public List<ColumnParams> columns() {
+        return columns;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        int id = catalog.objectIdGenState();
+
+        List<CatalogTableColumnDescriptor> viewColumns = columns.stream().map(CatalogUtils::fromParams).collect(toList());
+        CatalogSystemViewDescriptor descriptor = new CatalogSystemViewDescriptor(id, name, viewColumns);
+
+        return List.of(
+                new NewSystemViewEntry(descriptor),
+                new ObjectIdGenUpdateEntry(1)
+        );
+    }
+
+    private void validate() {
+        validateIdentifier(name, "Name of the system view");
+
+        if (nullOrEmpty(columns)) {
+            throw new CatalogValidationException("System view should have at least one column");
+        }
+
+        Set<String> columnNames = new HashSet<>();
+
+        for (ColumnParams column : columns) {
+            validateColumnParams(column);
+
+            if (!columnNames.add(column.name())) {
+                throw new CatalogValidationException(Common.INTERNAL_ERR,
```

Comment:
it's better to throw Catalog.VALIDATION_ERR

Reply:
Fixed.